src/collective/upload/browser.py

```diff
@@ -211,13 +212,15 @@ def render(self):
 
     def registry_config(self):
         config = {}
-        registry = getUtility(IRegistry)
-        settings = registry.forInterface(IUploadSettings)
-        config['extensions'] = str(settings.upload_extensions.replace(' ', '').replace(',', '|'))
-        config['max_file_size'] = settings.max_file_size
-        config['resize_max_width'] = settings.resize_max_width
-        config['resize_max_height'] = settings.resize_max_height
-
+        record = dict(name='upload_extensions', interface=IUploadSettings)
```

Comment:
this is way more complex and harder to read that it was before; I'm -1 on this change.